TeachingRecordSystem/src/TeachingRecordSystem.AuthorizeAccess/IdDbContext.cs

```diff
@@ -28,3 +30,84 @@ public class IdTrnToken
     [Column("user_id")]
     public Guid? UserId { get; set; }
 }
+
+[Table("users")]
+public class User
+{
+    [Column("user_id")]
+    public Guid UserId { get; set; }
+    [Column("email_address")]
+    public required string EmailAddress { get; set; }
+    [Column("first_name")]
+    public required string FirstName { get; set; }
+    [Column("middle_name")]
+    public string? MiddleName { get; set; }
+    [Column("last_name")]
+    public required string LastName { get; set; }
+    [Column("preferred_name")]
+    public string? PreferredName { get; set; }
+    [Column("date_of_birth")]
+    public DateOnly? DateOfBirth { get; set; }
+    [Column("created")]
+    public required DateTime Created { get; set; }
+    [Column("updated")]
+    public required DateTime Updated { get; set; }
+    [Column("last_signed_in")]
+    public DateTime? LastSignedIn { get; set; }
+    [Column("completed_trn_lookup")]
+    public DateTime? CompletedTrnLookup { get; set; }
+    [Column("user_type")]
+    public UserType UserType { get; set; }
+    [Column("trn")]
+    public string? Trn { get; set; }
+    [Column("trn_association_source")]
+    public TrnAssociationSource? TrnAssociationSource { get; set; }
+    [Column("staff_roles")]
+    public string[] StaffRoles { get; set; } = Array.Empty<string>();
+    [Column("registered_with_client_id")]
+    public string? RegisteredWithClientId { get; set; }
+    [Column("trn_lookup_status")]
+    public TrnLookupStatus? TrnLookupStatus { get; set; }
+    [Column("is_deleted")]
+    public bool IsDeleted { get; set; }
+    [Column("merged_with_user_id")]
+    public Guid? MergedWithUserId { get; set; }
+    [Column("mobile_number")]
+    public string? MobileNumber { get; set; }
+    [Column("trn_lookup_support_ticket_created")]
+    public bool TrnLookupSupportTicketCreated { get; set; }
+    [Column("trn_verification_level")]
+    public TrnVerificationLevel? TrnVerificationLevel { get; set; }
+    [Column("national_insurance_number")]
+    public string? NationalInsuranceNumber { get; set; }
+}
+
+public enum UserType
+{
+    Default = 0,
+    Teacher = 0,
+    Staff = 1
+}
+
+public enum TrnAssociationSource
+{
+    Lookup = 0,
+    Api = 1,
+    SupportUi = 2,
+    UserImport = 3,
+    TrnToken = 4,
+}
+
+public enum TrnLookupStatus
+{
+    None = 0,
+    Pending = 1,
+    Found = 2,
+    Failed = 3
+}
+
+public enum TrnVerificationLevel
+{
+    Low = 0,
+    Medium = 1
+}
```

Comment:
Can we wrap these inside a static class or child namespace so we don't 'pollute' the root namespace?

```
public static class IdModelTypes
{
    public enum UserType ...
}
```